test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignedPackageIntegrityVerificationTests.cs

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if IS_DESKTOP
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NuGet.Packaging.Signing;
+using NuGet.Test.Utility;
+using Test.Utility.Signing;
+using Xunit;
+
+namespace NuGet.Packaging.FuncTest
+{
+    [Collection("Signing Funtional Test Collection")]
+    public class SignedPackageIntegrityVerificationTests
+    {
+        private SigningSpecifications _specification => SigningSpecifications.V1;
+
+        private SigningTestFixture _testFixture;
+        private TrustedTestCert<TestCertificate> _trustedTestCert;
+        private IList<ISignatureVerificationProvider> _trustProviders;
+
+        public SignedPackageIntegrityVerificationTests(SigningTestFixture fixture)
+        {
+            _testFixture = fixture ?? throw new ArgumentNullException(nameof(fixture));
+            _trustedTestCert = _testFixture.TrustedTestCertificate;
+            _trustProviders = new List<ISignatureVerificationProvider>()
+            {
+                new IntegrityVerificationProvider()
+            };
+        }
+
+        [CIOnlyFact]
+        public async Task VerifyOnShiftedSignaturePackage_LastCDLastFEAsync()
```

Comment:
`*_WhenCentralDirectoryIsLastAndFileHeaderIsLastAsync()`?

Yeah, it's long but I couldn't tell what "FE" stood for.  [This](https://github.com/NuGet/NuGet.Client/pull/1853/files#diff-9a0802128eb2dc3cff9abbce0f9495f7R156) suggests it should be "LFH", but it doesn't hurt to expand it for readability.